### DISCUSSION TITLE
Move to TypedDict for kwargs

### DISCUSSION THREAD

**Discussion by davetapley** [pandas-dev/pandas-stubs#853] [Ideas]
Thoughts on using [PEP 692](https://peps.python.org/pep-0692/)'s `TypedDict` for kwargs [like this](https://stackoverflow.com/a/76466821/21115)?

I'm _guessing_ that would solve the problem I reported ⬇️ when using [`to_csv`](https://github.com/pandas-dev/pandas-stubs/blob/56eafc1ee579bfc3a145c71ec5deb0d5dd7b00ba/pandas-stubs/core/generic.pyi#L238-L286C11).
- https://github.com/microsoft/pyright/discussions/6995

**Comment by twoertwein** [comment]
Happy to have PRs that use `Unpack[_TypedDict]` for keywords that do not change between overloads (or even to share between functions such as read_csv and read_fwf) to make pandas-stubs more readable (also important for pandas itself).

xref https://github.com/PyCQA/flake8-pyi/issues/285

About your pyright issue: I think it is not working because you are not using a TypedDict. I don't think your pyright issue requires changes in pandas-stubs.

**Comment by Dr-Irv** [comment]
I like the idea, and would be open to PR's that use `TypedDict` and `Unpack`
